lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -75,6 +76,19 @@ export abstract class ClientApplication {
         return authorizationCodeClient.acquireToken(request);
     }
 
+    /**
+     * Acquires a token by exchanging the refresh token provided for a new set of tokens.
+     *
+     * This API is provided only for scenarios where you would like to migrate from ADAL to MSAL. Instead, it is
+     * recommended that you use acquireTokenSilent() for silent scenarios. When using acquireTokenSilent, MSAL will
+     * handle the caching and refreshing of tokens automatically.
+     * @param request
+     */
+    async acquireTokenByRefreshToken(request: RefreshTokenRequest): Promise<string>{
```

Comment:
Shouldn't this be acquireTokenSilent?

Reply:
acquireTokenSilent is different than acquireTokenByRefreshToken. 

acquireTokenSilent will check the cache, and 
1) finds an AT, return it
2) find an expired AT by valid RT, do acquireTokenByRefreshToken
3) if forceRefresh = true, cache is skipped and acquireTokenByRefreshToken

You can think of acquireTokenSilent as wrapping acquireTokenByRefreshToken. acquireTokenByRefreshToken is provided separately for migration scenarios

Reply:
So historically this has been wrapped in the acquireTokenSilent functionality. basically an application will call acquireTokenSilent and acquireTokenByRefreshToken will happen without the caller calling it explicitly. I haven't found any precedence in other MSAL libs for a separate API for this either, but if it is something Node specific I may be missing something.

Reply:
The three libraries that node is closest too (.NET, Java, Python) all implement it. We can chat about this at the next design meeting. 

For reference: 
 - [.NET](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/442227e1899ba45511840cbb1f59cfbd571b1546/src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs#L139)
- [Java](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/442227e1899ba45511840cbb1f59cfbd571b1546/src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs#L139) 
- [Python](https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal#migrate-existing-refresh-tokens-for-msal-python)

Note that: 
- Python doesn't expose this in their public API as people can just call directly into internal APIs, due to how Python access modifiers (or lack of) work.